smithy-openapi/src/test/resources/software/amazon/smithy/openapi/fromsmithy/examples-test.openapi.json

```diff
@@ -0,0 +1,520 @@
+                  "description": "fooTestDoc3",
+                  "value": {
+                    "fooBody1": "fooValidBody5",
+                    "fooBody2": "fooValidBody6"
+                  }
+                }
+              }
+            }
+          }
+        },
+        "parameters": [
+          {
+            "name": "fooLabel",
+            "in": "path",
+            "schema": {
+              "type": "string"
+            },
+            "required": true,
+            "examples": {
+              "Foo_PUT_example1": {
+                "summary": "Foo valid example",
+                "description": "fooTestDoc",
+                "value": "def"
+              },
+              "Foo_PUT_example2": {
+                "summary": "Foo error example",
+                "description": "fooTestDoc2",
+                "value": "klm"
+              },
+              "Foo_PUT_example3": {
+                "summary": "Foo error example 2",
+                "description": "fooTestDoc3",
+                "value": "qrs"
+              }
+            }
+          },
+          {
+            "name": "queryParam",
+            "in": "query",
+            "style": "form",
+            "schema": {
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            },
+            "explode": true,
+            "examples": {
+              "Foo_PUT_example1": {
+                "summary": "Foo valid example",
+                "description": "fooTestDoc",
+                "value": [
+                  "a",
+                  "b",
+                  "c"
+                ]
+              },
+              "Foo_PUT_example2": {
+                "summary": "Foo error example",
+                "description": "fooTestDoc2",
+                "value": [
+                  "x",
+                  "y",
+                  "z"
+                ]
+              },
+              "Foo_PUT_example3": {
+                "summary": "Foo error example 2",
+                "description": "fooTestDoc3",
+                "value": [
+                  "@",
+                  "#",
+                  "$"
+                ]
+              }
+            }
+          },
+          {
+            "name": "fooStringInputHeader",
+            "in": "header",
+            "schema": {
+              "type": "string"
+            },
+            "examples": {
+              "Foo_PUT_example1": {
+                "summary": "Foo valid example",
+                "description": "fooTestDoc",
+                "value": "abc"
+              },
+              "Foo_PUT_example2": {
+                "summary": "Foo error example",
+                "description": "fooTestDoc2",
+                "value": "hij"
+              },
+              "Foo_PUT_example3": {
+                "summary": "Foo error example 2",
+                "description": "fooTestDoc3",
+                "value": "nop"
+              }
```

Comment:
I think there should be some test case that shows the possibility of there being `Foo_example1` and `Foo_example3` but not `Foo_example2` for some location of the example, like maybe in this header. To validate that the keys follow the index in the examples trait, as opposed to the number incrementing based on the number of keys for that location.

Reply:
Test case added, and tweaked logic a bit to make it work as you've described it.